src/components/app.tsx

```diff
@@ -5,13 +5,43 @@ import { DragDropContext } from 'react-dnd';
 import HTML5Backend from 'react-dnd-html5-backend';
 import * as SplitPane from 'react-split-pane';
 
+import {Data, InlineData, isInlineData, isUrlData, UrlData} from 'vega-lite/build/src/data';
+
 import {DataPane} from './data-pane';
 import {EncodingPane} from './encoding-pane';
 import {Header} from './header';
 import {ViewPane} from './view-pane';
 
+import {
+  ActionHandler,
+  DatasetAsyncAction,
+  datasetUrlLoad,
+} from '../actions';
+
+
+export type VoyagerConfig = Object;
+export type VoyagerData = InlineData | UrlData;
+
+interface VoyagerAppProps extends ActionHandler<DatasetAsyncAction> {
+  config: Object;
+  data: Data;
```

Comment:
@tafsiri  I wonder if we could make this `data` named `fixedData`.
Basically, if there is a fixed dataset, we just don't show data selector and just show this fixedData.  Otherwise, show a data selector like normal.